cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -498,41 +498,37 @@ impl ProcessComposeLogReader {
                 let line = line.map_err(ServiceError::ReadLogLine)?;
 
                 // The receiver end was dropped, so we can't send any more messages.
-                // Might as well break out of the loop and kill the child process.
+                // Might as well kill the child process.
                 let Ok(_) = sender.send(ProcessComposeLogLine::new(&process, line)) else {
                     debug!("receiver dropped, stopping log reader");
-                    break;
+                    child.kill().map_err(ServiceError::ProcessComposeCmd)?;
+                    return Ok(());
                 };
             }
 
-            // Here, either sending to the receiver failed i.e. the receiver was dropped,
-            // or the child process died.
-            // If the child process ended, that's unexpected
-            // and we'll try to communicate that through the channel.
+            // Here, process-compose has closed stdout.
+            // It should have exited or be about to exit.
+            // kill() it just to be sure
+            child.kill().map_err(ServiceError::ProcessComposeCmd)?;
+
+            // Communicate why it exited through the channel.
             // The most likely error is that the socket doesn't exist,
             // trying to read logs for a non existent process
             // unfortunately just blocks indefinitely without any error message.
-
-            if let Some(exit_status) = child.try_wait().map_err(ServiceError::ProcessComposeCmd)? {
```

Comment:
I see this assumed the child process is already gone, and otherwise just kills it without inspecting its error

Reply:
Is that just an observation?

Reply:
yes